### PR TITLE
Fix UISIDetector grace period bug

### PR DESCRIPTION
**Pull request by BillCarsonFr**
## Type of change

- [x] Bugfix

## Content

Grace period for UISIDetection was not working properly, causing errors to be reported even if the event was decrypted during grace period.

Added Unit test on UISIDetector.

### CODE REVIEW

File: vector/src/test/java/im/vector/app/features/crypto/UISIDetectorTest.kt [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            }
+        }
+
+        // report a decryption error
+        val eventId = "0001"
+        val event = fakeEncryptedEvent(eventId, "s1", "r1")
+        uisiDetector.onEventDecryptionError(event, fakeCryptoError())
+
+        runBlocking {
+            delay(40_000)
```
Comment (bmarty) [inline vector/src/test/java/im/vector/app/features/crypto/UISIDetectorTest.kt:63]: The test will last at least 40 seconds? I think there is a way for test case like this much faster with coroutines (sort of fake time / accelerated time)

File: vector/src/test/java/im/vector/app/features/crypto/UISIDetectorTest.kt [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        uisiDetector.onEventDecryptionError(event, fakeCryptoError())
+
+        // the grace period is 30s
+        scope.launch(Dispatchers.Default) {
+            delay(10_000)
+            uisiDetector.onEventDecrypted(event, emptyMap())
+        }
+
+        runBlocking {
+            delay(60_000)
```
Comment (bmarty) [inline vector/src/test/java/im/vector/app/features/crypto/UISIDetectorTest.kt:97]: same

File: vector/src/test/java/im/vector/app/features/crypto/UISIDetectorTest.kt (hunk 1 above)
Comment (BillCarsonFr) [reply vector/src/test/java/im/vector/app/features/crypto/UISIDetectorTest.kt:63]: The grace period is 30s. I could make it configurable and then reduce the test duration, but is it really an issue if it's taking 40s?

Comment (bmarty) [reply vector/src/test/java/im/vector/app/features/crypto/UISIDetectorTest.kt:63]: we want test (and so the CI) to be fast!

Comment (BillCarsonFr) [reply vector/src/test/java/im/vector/app/features/crypto/UISIDetectorTest.kt]: I made the grace period configurable, and reduced the test duration.

File: vector/src/test/java/im/vector/app/features/crypto/UISIDetectorTest.kt (hunk 2 above)
Comment (BillCarsonFr) [reply vector/src/test/java/im/vector/app/features/crypto/UISIDetectorTest.kt]: done

**Review by bmarty — APPROVED** [review]
Thanks for the update.

**Review by Florian14 — APPROVED** [review]
LGTM

**Pull request merged**